src/main/java/org/elmlang/intellijplugin/ElmParserDefinition.java

```diff
@@ -21,12 +20,10 @@
 
 public class ElmParserDefinition implements ParserDefinition {
     public static final TokenSet WHITE_SPACES = TokenSet.create(TokenType.WHITE_SPACE);
+    // This is empty because we need to have everything available to the BNF grammar so that we can fold comments.
```

Comment:
This comment is out-of-date. It's no longer empty. And even if it were, I still struggled to understand its meaning.

As best as I can understand, you want to hide certain kinds of comments from the special behavior that IntelliJ does when it knows that a PsiElement is a comment. Seems like doing this will lead to unintended consequences down the road (but I'm not expert on IntelliJ plugin development).